pkg/readiness/object_tracker.go

```diff
@@ -257,19 +257,24 @@ func (t *objectTracker) Populated() bool {
 // Expectations must be populated before the tracker can be considered satisfied.
 // Expectations are marked as populated by calling ExpectationsDone().
 func (t *objectTracker) Satisfied() bool {
+	// Determine if we need access to the write lock, a precious resource
 	satisfied, needMutate := func() (bool, bool) {
 		t.mu.RLock()
 		defer t.mu.RUnlock()
 
-		// We'll only need the write lock in certain conditions:
-		//  1. We haven't tripped the circuit breaker
-		//  2. Expectations have been previously populated
-		//  3. We have expectations and observations - some of these may match can be resolved.
-		needMutate :=
-			!t.allSatisfied &&
-				t.populated &&
-				((len(t.seen) > 0 && len(t.expect) > 0) || // ...are there expectations we can resolve?
-					(len(t.seen) == 0 && len(t.expect) == 0)) // ...is the circuit-breaker ready to flip?
+		// matching observations and expectations may be able to be resolved
+		resolvableExpectations := len(t.seen) > 0 && len(t.expect) > 0
+
+		// We only need the write lock under certain conditions:
+		//  1. We haven't yet tripped the circuit breaker (t.allSatisfied)
+		//  2. We have received all necessary expectations (t.populated)
+		//  3. Certain expectation/observation states
```

Comment:
3 should read:

```
3. there is potential for action to be taken 
```

Also, make it clear that these conditions are and-ed. E.g.:

`We only need the write lock when all of the following are true:`

Reply:
Done